Add wasmJs target and support

EpicSquid opened the pull request:
Using Ktor 3.x and Kotlin 2.0 I've added a wasmJs target for publishing so it can be used fully cross platform.

EpicSquid committed:
Added WASM support (basic)

EpicSquid committed:
Fixed tab vs space indentation

Ingwersaft commented:
Hey, thanks for the contribution!

Looks like https://github.com/actions/upload-artifact/issues/691 broke the pipeline. Could you switch to `actions/upload-artifact@v4` in your branch please?

EpicSquid commented:
Sure thing, sorry haven't been able to get to this yet. Will try again this week

EpicSquid committed:
Updated upload-artifact@v4 to fix issues with the pipeline

EpicSquid committed:
Build now works

Updates include:
 - Gradle updated to use libs.version.toml for consistency with gradle standards
- Removed NPM publish plugin as it is no longer maintained and is incompatible with kotlin wasm
 - Removed JsExport annotations as they are not compatible with wasm (which has different export restrictions).
 - Updated some dependencies

EpicSquid commented:
I know I've made some changes that may need rectifying, but now the build works correctly I wanted to see if tests pass in the pipeline.

EpicSquid committed:
Added tests matching ios

Ingwersaft commented:
```
Anyone with collaborator access to this repository can use these secrets and variables for actions. They are not passed to workflows that are triggered by a pull request from a fork.
```
You can tell that I don't use github very often :D 

I pushed your changes so the pipeline runs with the secrets: #75

henwoods commented:
I was just looking to work on bumping versions to fix the [ktor compatiblity issues](https://github.com/Liftric/cognito-idp/issues/72) I'm also experience, but the work is already done here!

Any way we can get some movement on this 🙏

Ingwersaft commented:
https://github.com/Liftric/cognito-idp/pull/75/checks fails during `Build testapp and test request (iOS)`:
```
Testing failed:

Test session results, code coverage, and logs:
	/Users/runner/Library/Developer/Xcode/DerivedData/TestApp-bnlmepeuogcisibknrjooplcfvfa/Logs/Test/Run-TestApp-2025.05.22_12-22-07-+0000.xcresult

	Unable to Install “com.liftric.cognito.idp.TestApp”
	TestApp encountered an error (Failed to install or launch the test runner. (Underlying Error: Unable to Install “com.liftric.cognito.idp.TestApp”. This app cannot be installed because its integrity could not be verified. Failed to verify code signature of /var/db/appinstalld/Library/Caches/com.apple.mobile.installd.staging/temp.dnIRNG/extracted/TestApp.app : 0xe800801c (No code signature found.)))
```

As a linux user I can't debug it and no apple user at our place has time for it right now.

PruszynskiJakub commented:
It's probably related to profile provisioning. Although I am quite new on iOS. Could help @Ingwersaft ?
